blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -430,6 +450,26 @@ private MetricData getStringMetricDataForRange(Locator locator, Range range, Gra
         return new MetricData(points, getUnitString(locator), MetricData.Type.STRING);
     }
 
+    private MetricData getEnumMetricDataForRange(final Locator locator, Range range, Granularity gran, RollupType rollupType, DataType dataType) {
```

Comment:
Also, the code in this method is doing basically the same as the code above in the getDataToRoll() method.  Is there any way you can merge most of the functionality into the same method?